src/frontend/src/utils/iiConnection.test.ts

```diff
@@ -0,0 +1,103 @@
+import { MetadataMapV2, _SERVICE } from "$generated/internet_identity_types";
+import {
+  AnchorMetadata,
+  RECOVERY_PAGE_SHOW_TIMESTAMP_MILLIS,
+} from "$src/repositories/anchorMetadata";
+import { ActorSubclass } from "@dfinity/agent";
+import { DelegationIdentity } from "@dfinity/identity";
+import { AuthenticatedConnection } from "./iiConnection";
+import { MultiWebAuthnIdentity } from "./multiWebAuthnIdentity";
+
+const mockDelegationIdentity = {
+  getDelegation() {
+    return {
+      delegations: [],
+    };
+  },
+} as unknown as DelegationIdentity;
+
+const recoveryPageShownTimestampMillis = 1234567890;
+const mockRawMetadata: MetadataMapV2 = [
+  [
+    RECOVERY_PAGE_SHOW_TIMESTAMP_MILLIS,
+    { String: String(recoveryPageShownTimestampMillis) },
+  ],
+];
+const mockAnchorMetadata: AnchorMetadata = {
+  recoveryPageShownTimestampMillis,
+};
+const expectedAnchorInfo = {
+  devices: [],
+  device_registration: [],
+};
+
+// Used to await that the getter has resolved.
+let infoResponse: MetadataMapV2 | null | undefined = null;
+
+const mockActor = {
+  get_anchor_info: vi.fn().mockResolvedValue(expectedAnchorInfo),
+  identity_info: vi.fn().mockImplementation(async () => {
+    // The `await` is necessary to make sure that the `getterResponse` is set before the test continues.
+    infoResponse = await mockRawMetadata;
+    return { Ok: { metadata: mockRawMetadata } };
+  }),
+  identity_metadata_replace: vi.fn().mockResolvedValue({ Ok: null }),
+} as unknown as ActorSubclass<_SERVICE>;
+
+test("gets anchor info from actor", async () => {
```

Comment:
While it is good to test things, I'm a bit confused as to how `getAnchorInfo` relates to the identity metadata functionality added in this PR?

Reply:
I was trying that I could successfully test AuthenticatedConnection and then I thought it might be ok to leave it.

But let's do that in another PR if I want to add more tests. I removed it from this PR.